### DISCUSSION TITLE
Environment variables not getting detected.

### DISCUSSION THREAD

**Discussion by driedpampas** [honojs/hono#3183] [Q&A]
Trying to get a secret key named "SECRET" (added with wrangler) gives this error: `Cannot destructure property 'SECRET' of 'Ks(...)' as it is undefined.` 
When running `bun dev` the error turns into `Cannot read properties of undefined (reading 'raw')`. The SECRET key is defined in `.env`.

```
    const { SECRET } = env(c)
    const token = await getSignedCookie(c, SECRET, 'token');
    const isUser = !!token;
    return isUser;
```
---
Trying to use it like this: 
```
    const token = await getSignedCookie(c, c.env.SECRET, 'token');
    const isUser = !!token;
    return isUser;
```
returns `Cannot read properties of undefined (reading 'SECRET')`

**Comment by nakasyou** [comment]
Hi @driedpampas.

You said you use wrangler, wrangler doesn't load environment variables from `.env`.
Try `.dev.vars`.
https://developers.cloudflare.com/workers/configuration/environment-variables/

**Reply by driedpampas** [reply]
I tried that too but in Dev mode it made no difference (i didn't think it would since `bun dev` just runs `vite` which reloads itself only when editing `.env`). And in production it should be passed by the Pages environment anyway:
> The secret’s value is passed through as defined.
